tests/integration/test_random_creation.py

```diff
@@ -33,6 +39,447 @@ def test_randn():
+    arr_np, arr_num = gen_random_from_both("random_sample", size=None)
+    assert np.isscalar(arr_np) == num.isscalar(arr_num)
+
+
+@pytest.mark.parametrize("size", LARGE_RNG_SIZES, ids=str)
+def test_random_sample(size):
+    arr_np, arr_num = gen_random_from_both("random_sample", size=size)
+    assert_distribution(
+        arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+    )
+
+
+class TestRandomErrors:
+    def assert_exc_from_both(self, func, exc, *args, **kwargs):
+        with pytest.raises(exc):
+            getattr(np.random, func)(*args, **kwargs)
+        with pytest.raises(exc):
+            getattr(num.random, func)(*args, **kwargs)
+
+    @pytest.mark.parametrize(
+        "seed, expected_exc",
+        [
+            (-100, ValueError),
+            (12.0, TypeError),
+            ("abc", TypeError),
+        ],
+        ids=lambda x: f" {str(getattr(x, 'expected_exception', x))} ",
+    )
+    @pytest.mark.xfail(reason="NumPy raises exceptions, cuNumeric pass")
+    def test_invalid_seed(self, seed, expected_exc):
+        self.assert_exc_from_both("seed", expected_exc, seed)
+        # -100: NumPy raises ValueError: Seed must be between 0 and 2**32 - 1
+        # 12.0: NumPy raises TypeError: Cannot cast scalar from
+        # dtype('float64') to dtype('int64') according to the rule 'safe'
+        # "abc": TypeError: Cannot cast scalar from dtype('<U3') to
+        # dtype('int64') according to the rule 'safe'
+        # cuNumeric accepts both -100 and 12.0, raises ValueError on "abc"
+        # ValueError: invalid literal for int() with base 10: 'abc'
+
+    @pytest.mark.parametrize(
+        "shape, expected_exc",
+        [
+            (-100, ValueError),
+            (12.0, TypeError),
+            ("abc", TypeError),
+            (None, TypeError),
+            ((12345,), TypeError),
+        ],
+        ids=lambda x: f" {str(getattr(x, 'expected_exception', x))} ",
+    )
+    def test_rand_invalid_shape(self, shape, expected_exc):
+        self.assert_exc_from_both("rand", expected_exc, shape)
+
+    @pytest.mark.xfail(
+        LEGATE_TEST and not EAGER_TEST,
+        reason="cuNumeric does not raise error when LEGATE_TEST=1",
+    )
+    @pytest.mark.parametrize(
+        "low, high, expected_exc",
+        [
+            pytest.param(
+                -10000,
+                None,
+                ValueError,
+            ),
+            pytest.param(
+                -10000,
+                -20000,
+                ValueError,
+            ),
+        ],
+        ids=str,
+    )
+    def test_randint_invalid_range(self, low, high, expected_exc):
+        # When LEGATE_TEST=1:
+        # (-10000, None):
+        # NumPy & cuNumeric LEGATE_TEST=0: ValueError: high <= 0
+        # cuNumeric LEGATE_TEST=1: array([3124366888496675138])
+        # (-10000, -20000):
+        # NumPy & cuNumeric LEGATE_TEST=0: ValueError: low >= high
+        # cuNumeric LEGATE_TEST=1: array([-5410197118219848280])
```

Comment:
@manopapad thoughts on the difference with `LEGATE_TEST` set?

Reply:
Size checks are missing; added with https://github.com/nv-legate/cunumeric/pull/966.